src/blocks/data.js

```diff
@@ -680,6 +699,67 @@ class Scratch3DataBlocks {
             }
         );
     }
+
+    renameCallback ({blockInfo, newName, varType}) {
```

Comment:
I wasn't sure about whether to pass in varType here or determine it based on the opcode. See related code in the GUI pr which mentions having access to the VM's `Variable` class.